### DISCUSSION TITLE
Dynamically set measurement range in runtime

### DISCUSSION THREAD

**Discussion by Teslafly** [google/openhtf#1167] [Q&A] [answered]
I need to dynamically set the acceptable validation range of a measurement in run time. so I can't use a measurement decorator to pre-set this value. none of the example code covers this use case. Is there a best way to do this? 

Ideally I want a function I call with the measurement constructor and value to log and validate everything at once
`test.log_measurement(value, htf.Measurement("3p3v_rail").with_validator(validators.InRange(3.2, 3.4), eval_immediately=True)`

I did find this "dynamic measurements" section in the spintop openhtf documentation, but spintop is highly diverged and not maintained.
https://spintop-openhtf.readthedocs.io/_/downloads/en/latest/pdf/#section.11.5

It also seems pretty hacky. directly touching the test sequence state.

**Comment by glados-verma** [comment, marked as answer]
Modifying the measurement specification after the test has started is not supported in OpenHTF. The Diagnoses feature was added relatively recently but I'm not sure if it covers your use case exactly, not having used it myself yet.

Before Diagnoses, the typical workaround was to have a boolean measurement, something like "<my measurement>_pass", and keep your other measurements without validators. At runtime, check the values and set the boolean measurement.
